documentation/breaking-changes/breaking-changes-in-generated-modules-due-to-codegen-tool-upgrade-from-v3-to-v4.md

```diff
@@ -0,0 +1,42 @@
+# Breaking Changes in Generated Modules Due to AutoRest.PowerShell Upgrade from v3 to v4
+
+We are moving forward with using the new version of the code generation tool, AutoRest.PowerShell v4, to take advantage of the features offered by the latest version. The redesign of the generated code to support new user requirements in this version inevitably introduces some breaking changes.
+
+Below, we list the potential breaking changes from the perspective of users that may occur when upgrading the code generation tool for Azure PowerShell modules.
+
+**Note: The breaking changes listed below may not apply universally to all generated modules. The occurrence of these breaking changes also depends on the specific configurations used when generating Azure PowerShell modules with AutoRest.PowerShell v4. For detailed information on which breaking changes may occur, please refer to the [breaking change list](https://learn.microsoft.com/en-us/powershell/azure/upcoming-breaking-changes) for the specific module.**
+
+## List Replaces Array in Generated C# Classes
+
+We use `List` to replace `Array` mainly because `List` is a dynamic array that allows adding new elements using the Add method.
```

Comment:
```suggestion
We now generate collection-like properties as [`List`](https://learn.microsoft.com/dotnet/api/system.collections.generic.list-1) instead of [`Array`](https://learn.microsoft.com/en-us/dotnet/api/system.array) in the models. This is because `List` is dynamic that it allows appending/deleting elements, while `Array` is fixed-size hence difficult to edit.
```